core/zip/test/ZipTest.cxx

```diff
@@ -0,0 +1,66 @@
+#include <Compression.h>
+#include <RZip.h>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+static void testZipBufferSizes(ROOT::RCompressionSetting::EAlgorithm::EValues compressionAlgorithm)
+{
+   static constexpr size_t BufferSize = 256;
+   static constexpr size_t MaxBytes = 128;
+   static_assert(MaxBytes <= BufferSize, "MaxBytes must be smaller than BufferSize");
+   // For extra "safety", allocate the buffers on the heap to avoid corrupting the stack should anything go wrong.
```

Comment:
In addition, I would recommendation to over-allocate and tain the before and after to insure that there is no over-write (even for the larger sizes)
```
   std::unique_ptr<char[]> sourceBuffer(new char[BufferSize+2 * 8]);
   static_cast<unsigned long*>( &source[0]) = 0xdeadbeef;
   static_cast<unsigned long*>( &source[BufferSize+1]) = 0xdeadbeef;
   char *source = &(source[4]);
```

Reply:
This is already exactly how the test is working: The buffers have size `BufferSize`, but the code scans from 1 to `MaxBytes` bytes. All bytes are initially filled with monotonically increasing numbers, which can be checked starting from the byte offset that compression is not supposed to touch, in every iteration. I don't really see a value in tainting the memory before the buffers, what would this protect against?

Reply:
humm ... it is 'easy' to add or so it seems.  See https://github.com/root-project/root/pull/14523/files#r1481944149

Reply:
Ok, I did the slightly more complicated setup of moving the `MaxBytes` exactly in the middle.